bridge-token-factory/src/lib.rs

```diff
@@ -378,6 +395,50 @@ impl BridgeTokenFactory {
             Balance::from(current_storage - initial_storage) * STORAGE_PRICE_PER_BYTE;
         required_deposit
     }
+
+    pub fn set_metadata(
+        &self,
+        address: String,
+        name: Option<String>,
+        symbol: Option<String>,
+        reference: Option<String>,
+        reference_hash: Option<Base64VecU8>,
+        decimals: Option<u8>,
+        icon: Option<String>,
+    ) -> Promise {
+        assert!(self.controller_or_self());
+        ext_bridge_token::set_metadata(
+            name,
+            symbol,
+            reference,
+            reference_hash,
+            decimals,
+            icon,
+            &self.get_bridge_token_account_id(address),
+            env::attached_deposit(),
+            env::prepaid_gas() - OUTER_SET_METADATA_GAS,
+        )
+    }
+
+    pub fn get_controller(&self) -> Option<AccountId> {
+        env::storage_read(CONTROLLER_STORAGE_KEY)
+            .map(|value| String::from_utf8(value).expect("Invalid controller account id"))
+    }
+
+    pub fn set_controller(&mut self, controller: AccountId) {
+        assert_self();
+        assert!(env::is_valid_account_id(controller.as_bytes()));
+        env::storage_write(CONTROLLER_STORAGE_KEY, controller.as_bytes());
+    }
+
+    pub fn controller_or_self(&self) -> bool {
+        let caller = env::predecessor_account_id();
+        caller == env::current_account_id()
+            || self
+                .get_controller()
+                .map(|controller| controller == caller)
+                .unwrap_or_default()
```

Comment:
`.unwrap_or_default()` is dangerous here, because in case if default value will be true for some reason, this will be evaluated as `caller == env::current_account_id() || true` which will be `true`.

So I suggest using `.unwrap_or(false)`